test/Microsoft.AspNet.Mvc.Razor.Test/Compilation/CompilerCacheTest.cs

```diff
@@ -443,6 +444,67 @@ public void ConstructorNormalizesPrecompiledViewPath(string viewPath)
             Assert.Same(result1.CompilationResult.CompiledType, result2.CompilationResult.CompiledType);
         }
 
+        [Fact]
+        public void GetOrAdd_CachesCompilationExceptions()
+        {
+            // Arrange
+            var fileProvider = new TestFileProvider();
+            fileProvider.AddFile(ViewPath, "some content");
+            var cache = new CompilerCache(fileProvider);
+
+            // Act and Assert - 1
+            Assert.Throws<InvalidTimeZoneException>(() =>
```

Comment:
This looks to be more apart of the Arrange section. `Assert.Throws<>(throw new Exception)` seems odd to assert.